src/main/java/com/google/cloud/resourcemanager/Lien.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.resourcemanager;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.gax.paging.Page;
+import com.google.cloud.resourcemanager.ResourceManager.LienListOption;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A Google Cloud Resource Manager lien object.
+ *
+ * <p>A Lien represents an encumbrance on the actions that can be performed on a resource. This is
+ * an immutable class. Methods that change or update the underlying Lien information return a new
+ * Lien instance. {@code Lien} adds a layer of service-related functionality over {@link LienInfo}.
+ */
+public class Lien extends LienInfo {
+
+  private final ResourceManagerOptions options;
+  private transient ResourceManager resourceManager;
+
+  /** Builder for {@code Lien}. */
+  public static class Builder extends LienInfo.Builder {
```

Comment:
This class would be easier to read if this Builder was not an inner class.

Reply:
We required to have this builder for to create an instance of lien with object of resource manager.

Reply:
Having a builder is fine, but this could be an outer class.

Reply:
could be but As per Builder Design Pattern always Builder kept as an inner class.

Reply:
No, the Builder design pattern works perfectly well without inner classes. Indeed it's fully appropriate for languages and versions of Java that do not have inner classes.

Reply:
please check the existing implementation of this `Resourcemanager Client`.

Reply:
I'm not sure what you're referring to...

Reply:
already simplified code